source/javascripts/directives/image_loader.coffee

```diff
@@ -0,0 +1,20 @@
+EGA.directive 'awardHeader', ->
+  restrict: 'E'
+  scope:
+    award: '='
+  transclude: true
+  replace: true
+  template: "<header ng-transclude></header>"
+  link: (scope, element, attrs) ->
+    scope.$watch 'award', (award) ->
+      images = award.images
+
+      loadImage = (index = 0) ->
+        return if index >= images.length
+
+        imageObj = new Image
+        imageObj.onload = -> element.prepend(imageObj)
+        imageObj.onerror = -> loadImage(index + 1)
+        imageObj.src = images[index]
+
+      loadImage()
```

Comment:
Not exactly, I wouldn't have called it `awardHeader` because that's only related to this project. I would leave the `<header>` html tag in the `index.erb` and put this directive inside. It should have its own `<div>` as root element and inside its either an image or the alternative content.

But for this project it doesn't matter, so let's keep it like that. Now it would be YAGNI to change it ;-)